agents/plugins/mk_mysql

```diff
@@ -40,6 +40,20 @@ do_query() {
 
 }
 
+if ! type mysqladmin >/dev/null 2>&1; then
+    # Try some default mysql client binary paths if they aren't in $PATH already
+    # HINT: If yours is not in the list you may add a symlink with "ln -s /usr/local/mysql <your installation path>"
+    if [ -e "/usr/local/mysql/bin/mysqladmin" ]; then
+        PATH=${PATH}:/usr/local/mysql/bin/
```

Comment:
In our 2.1 release we had a problem with an indefinitely growing path variable in our async agent. The problem was exactly a line like this, where we always append to the PATH. Turns out if your PATH variable is to long you cannot run any program under linux. This is a hard to debug issue if it occurs.

Currently plugins are called as sub-shells and this is not a problem. I would prefer if we do not have to rely on that assumption. Instead of changing path can you try to create a mysqladmin function like this.

`mysqladmin() { /usr/local/mysql/bin/mysqladmin "$@"; }`

If that does not work here is our current trick to only update the path variable if a given string is not present in it.
https://github.com/tribe29/checkmk/blob/master/agents/check_mk_agent.linux#L132

Reply:
I'm thinking of settign a MYSQL_BINDIR variable and prefixing this to the mysql binary calls instead of appending the PATH variable. What do you think about that idea?

Reply:
sounds good too.